tests/conftest.py

```diff
@@ -48,39 +80,252 @@ def swg_config(indexd_client): # noqa
     config.host = indexd_client.url
     config.username = indexd_client.auth[0]
     config.password = indexd_client.auth[1]
-    yield config
+    return config
+
+
+@pytest.fixture
+def swg_config_no_migrate(indexd_client_no_migrate): # noqa
+    config = swagger_client.Configuration()
+    config.host = indexd_client_no_migrate.url
+    config.username = indexd_client_no_migrate.auth[0]
+    config.password = indexd_client_no_migrate.auth[1]
+    return config
 
 
 @pytest.fixture
 def swg_index_client(swg_config):
-    api = swagger_client.IndexApi(swagger_client.ApiClient(swg_config))
-    yield api
+    return swagger_client.IndexApi(swagger_client.ApiClient(swg_config))
+
+
+@pytest.fixture
+def swg_index_client_no_migrate(swg_config_no_migrate):
+    return swagger_client.IndexApi(swagger_client.ApiClient(swg_config_no_migrate))
 
 
 @pytest.fixture
 def swg_global_client(swg_config):
-    api = swagger_client.GlobalApi(swagger_client.ApiClient(swg_config))
-    yield api
+    return swagger_client.GlobalApi(swagger_client.ApiClient(swg_config))
+
+
+@pytest.fixture
+def swg_global_client_no_migrate(swg_config):
+    return swagger_client.GlobalApi(swagger_client.ApiClient(swg_config_no_migrate))
 
 
 @pytest.fixture
 def swg_alias_client(swg_config):
-    api = swagger_client.AliasApi(swagger_client.ApiClient(swg_config))
-    yield api
+    return swagger_client.AliasApi(swagger_client.ApiClient(swg_config))
+
+
+@pytest.fixture
+def swg_alias_client_no_migrate(swg_config):
+    return swagger_client.AliasApi(swagger_client.ApiClient(swg_config_no_migrate))
 
 
 @pytest.fixture
 def swg_dos_client(swg_config):
-    api = swagger_client.DOSApi(swagger_client.ApiClient(swg_config))
-    yield api
+    return swagger_client.DOSApi(swagger_client.ApiClient(swg_config))
+
+
+@pytest.fixture
+def swg_dos_client_no_migrate(swg_config):
+    return swagger_client.DOSApi(swagger_client.ApiClient(swg_config_no_migrate))
 
 
 @pytest.fixture
 def swg_query_client(swg_config):
-    api = swagger_client.QueryApi(swagger_client.ApiClient(swg_config))
-    yield api
+    return swagger_client.QueryApi(swagger_client.ApiClient(swg_config))
+
+
+@pytest.fixture
+def swg_query_client_no_migrate(swg_config):
+    return swagger_client.QueryApi(swagger_client.ApiClient(swg_config_no_migrate))
+
+
+@pytest.fixture
+def swg_bulk_client(swg_config):
+    return swagger_client.BulkApi(swagger_client.ApiClient(swg_config))
+
+
+@pytest.fixture
+def swg_bulk_client_no_migrate(swg_config):
+    return swagger_client.BulkApi(swagger_client.ApiClient(swg_config_no_migrate))
+
+
+@pytest.fixture
+def index_driver():
+    driver = SQLAlchemyIndexDriver(PG_URL)
+    yield driver
+    drop_tables(driver, index_base)
+    driver.dispose()
+
+
+@pytest.fixture
+def alias_driver():
+    driver = SQLAlchemyAliasDriver(PG_URL)
```

Comment:
Ideally PG_URL is actually a fixture it these fixtures depend on them? Probably returned from the database setup right?

Reply:
PG_URL is just a string declared at the top of the same file. I'm not sure it needs to be a pytest fixture but it can be if you feel it's necessary.

https://github.com/NCI-GDC/indexd/blob/feat/jsonb/tests/conftest.py#L23